src/io/hdf5/hdf5_writer.cpp

```diff
@@ -19,16 +19,16 @@ namespace dca {
 namespace io {
 // dca::io::
 
+bool fileExists(const std::string& filename) {
+  std::ifstream ifile(filename);
+  return bool(ifile);
+}
+
 HDF5Writer::~HDF5Writer() {
   if (file_)
     close_file();
 }
 
-bool HDF5Writer::fexists(const char* filename) {
```

Comment:
This should be kept, checking for an ADIOS2 "file" should be through the specialized interface still.

Reply:
file exist is not a property of an hdf5 object, as it just checks if a file exists or not it should need no specialization for a different writer. I admit that the location of the free function is awkward tough, we should just use `std::filesystem::exists`, do you agree?